NTupleMaker/interface/functionsCP.h

```diff
@@ -34,215 +35,184 @@ TLorentzVector gen_ipVec(const AC1B * analysisTree, int tauIndex, int piIndex, T
+
   TLorentzVector tau2Prong;
   tau2Prong=chargedPivec(analysisTree,tauIndex2);
+  int piIndexfortau2=chargedPiIndex(analysisTree,tauIndex2);
+  if(piIndexfortau2>-1){
+    otree->VxConstitTau2=analysisTree->tau_constituents_vx[tauIndex2][piIndexfortau2];
+    otree->VyConstitTau2=analysisTree->tau_constituents_vy[tauIndex2][piIndexfortau2];   
+    otree->VzConstitTau2=analysisTree->tau_constituents_vz[tauIndex2][piIndexfortau2];      
+  }
+    
   
-  bool firstNegative = false;
-  if (analysisTree->tau_charge[tauIndex1]<0.0)
+  bool firstNegative = false;//Merijn 2019 1 10: instead I ask if the second is positive. WON'T WORK FOR E-MU CASE!
+//comment 2019 2 5: I need to re-evaluate why I implemented this. I think it is due to fact that initially lepton and muon index were innertwined
+  if (analysisTree->tau_charge[tauIndex2]>0.0)
     firstNegative = true;
 
   TLorentzVector tau1IP;
-//Merijn: for leptonic decay calculate in different way than for hadronic
+  //Merijn: for leptonic decay calculate in different way than for hadronic
   if(channel=="et"||channel=="mt") tau1IP = ipVec_Lepton(analysisTree,tauIndex1,channel);
-  else{tau1IP = ipVec(analysisTree,tauIndex1);}//tt
+  else{tau1IP = ipVec(analysisTree,tauIndex1);}//tt: treat as tau index..
   TLorentzVector tau1Pi0;
   tau1Pi0.SetXYZT(0.,0.,0.,0.);
-  
+
+  //Merijn: have to assume here we WON'T look into e-mu case! 
   TLorentzVector tau2IP;
   tau2IP = ipVec(analysisTree,tauIndex2);
   TLorentzVector tau2Pi0;
-  tau2Pi0.SetXYZT(0.,0.,0.,0.);
+  tau2Pi0.SetXYZT(0.,0.,0.,0.); 
 
   //Merijn: here calculate neutral pion vectors. Only look for pions in first tau if channel is tt.
-  if(channel=="tt"){ if (analysisTree->tau_decayMode[tauIndex1]>=1&&analysisTree->tau_decayMode[tauIndex1]<=3) tau1Pi0 = neutralPivec(analysisTree,tauIndex1);}
-  if (analysisTree->tau_decayMode[tauIndex2]>=1&&analysisTree->tau_decayMode[tauIndex2]<=3) tau2Pi0 = neutralPivec(analysisTree,tauIndex2);
+  if(channel=="tt"){ if(analysisTree->tau_decayMode[tauIndex1]>=1&&analysisTree->tau_decayMode[tauIndex1]<=3) tau1Pi0 = neutralPivec(analysisTree,tauIndex1);}
+
+  if (analysisTree->tau_decayMode[tauIndex2]>=1&&analysisTree->tau_decayMode[tauIndex2]<=3){ 
+	tau2Pi0 = neutralPivec(analysisTree,tauIndex2);
+  }
   
-//here can proceed with CP calculation
-  otree->acotautau_00=acoCPCOUT(tau1Prong,tau2Prong,tau1IP,tau2IP,firstNegative,false,false); // cout<<"otree->acotautau_00 "<<otree->acotautau_00<<endl;
 
-  if (analysisTree->tau_decayMode[tauIndex1]==1&&analysisTree->tau_decayMode[tauIndex2]==0)
-    otree->acotautau_10=acoCP(tau1Prong,tau2Prong,tau1Pi0,tau2IP,firstNegative,true,false);
+  otree->acotautau_00=acoCP(tau1Prong,tau2Prong,tau1IP,tau2IP,firstNegative,false,false,otree);
+   //I think it should work since everything assigned for 3 cases. Note: aco_00 will be filled with mu x 1-prong and mu x 1.1-prong
```

Comment:
Please try to rephrase that, I don't understand what you mean by "mu x 1.1-prong"

Reply:
Hi this is indeed and "internal comment".  1.1 prong is just a short way of putting "charged plus neutral pion", which is always a bit of a  mouthful.
But perhaps we could use 1 prong and rho-prong, this would be easier also for potential future users..

Reply:
I would avoid calling it rho-prong...
When you take into account the a1 resonance you have both the charged pi + 2 pi0s, and the 3 charged pi decay channels. The former decay is treated exactly as the rho resonance, since we don't count the number of pi0s in the decay, but just sum all the electrons, positrons and photons that come from the decay of the pi0s.
Anyway it's a matter of notation, I was just curious for what you meant.